components/gitpod-db/src/typeorm/webhook-event-db-impl.ts

```diff
@@ -52,19 +52,23 @@ export class WebhookEventDBImpl implements WebhookEventDB {
         return query.getMany();
     }
 
-    public async deleteOldEvents(ageInDays: number, limit?: number): Promise<void> {
+    public async deleteOldEvents(ageInDays: number, limit: number): Promise<void> {
         const repo = await this.getRepo();
         const d = new Date();
         d.setDate(d.getDate() - ageInDays);
         const expirationDate = d.toISOString();
-        const query = repo
-            .createQueryBuilder()
-            .update()
-            .set({ deleted: true })
-            .where("creationTime <= :expirationDate", { expirationDate });
-        if (typeof limit === "number") {
-            query.limit(limit);
-        }
-        await query.execute();
+        await repo.query(
+            `DELETE FROM d_b_webhook_event
+                WHERE id IN (
+                SELECT id FROM (
```

Comment:
The subselect feels overly complicated. Is [the problem we're seeing here](https://github.com/gitpod-io/gitpod/pull/17650#discussion_r1197578844) with Typeorm?

Then we should do `DELETE FROM d_b_webhook_event WHERE creationTime <= ? LIMIT ?` I think.
Am I missing something? :thinking:

Reply:
It was based on this https://stackoverflow.com/questions/7142097/mysql-delete-statement-with-limit

Reply:
That question is about offsets (` LIMIT 20, 50`).

Reply:
updated